File: PRACTICAS/PRACTICA_ACTUAL/JAVA/JAVA/Civitas_P1/src/civitas_p1/CasillaImpuesto.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas_p1;

import java.util.ArrayList;

/**
 *
 * @author javierramirezp
 */
public class CasillaImpuesto extends CasillaDescanso{
    private float importe;
    
    CasillaImpuesto(String nombre, float importe){
    
        super(nombre);
        this.importe=importe;
    
    }
    
    @Override
    void recibeJugador(int iactual, ArrayList<Jugador> todos){
        if (jugadorCorrecto(iactual, todos)){
            informe(iactual, todos);
            todos.get(iactual).pagaImpuesto(importe);
        }
    }
    @Override
    public String toString(){   
        return " llamada " + nombre + " con un valor de "+importe+ " €";
    }
}

File: PRACTICAS/PRACTICA_ACTUAL/RUBY/pkg/Civitas-0.0.1/lib/casilla_impuesto.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require_relative 'casilla.rb'
require_relative 'jugador.rb'

module Civitas
  class Casilla_impuesto < Casilla
    
    attr_reader :importe
    
    def initialize(nombre,importe)
      super(nombre)
      @importe = importe
    end

     def recibe_jugador(iactual,todos)
      if (jugador_correcto(iactual,todos))
          informe(iactual,todos)
        todos[iactual].paga_impuesto(@importe)
      end
    end
    
     #Override
     def to_s
      puts "Tipo Casilla Impuesto, importe: #{@importe}"
    end

  end
end